keyboards/lily58/keymaps/lily58l_vii33/keymap.c

```diff
@@ -0,0 +1,225 @@
+  XXXXXXX,  KC_BSPC, QC_CTLX, QC_CTLC, QC_CTLV, XXXXXXX, _______,    _______,  XXXXXXX, KC_HOME, KC_BSPC, KC_END,   XXXXXXX, KC_ENT, 
+                              _______, _______, _______, _______,    KC_LSFT,  _______, _______, _______
+),
+
+
+[_UPPER] = LAYOUT(
+  _______,  _______,  _______,  _______,  _______,  _______,                    _______,  _______,  _______,  _______, _______, _______, 
+  _______,  QC_AT,    KC_TILD,  QC_EUR,   KC_PSCR,  _______,                    KC_PAST,  KC_LT,    KC_LCBR,  KC_RCBR, KC_PMNS, _______,
+  KC_PIPE,  XXXXXXX,  QC_SS,    QC_UNDO,  QC_REDO,  CTALDEL,                    KC_PSLS,  KC_EQL,   KC_LPRN,  KC_RPRN, KC_PPLS, QC_DQUT, 
+  QC_DEG,   KC_BSPC,  QC_CTLX,  QC_CTLC,  QC_CTLV,  XXXXXXX, _______, _______,  QC_HAT,   KC_GT,    KC_LBRC,  KC_RBRC, QC_AND,  KC_PENT,
+                                _______,  _______,  _______, KC_LSFT, _______,  _______,  _______,  _______
+),
+
+
+  [_MOD] = LAYOUT(
+    RGB_TOG, RGB_M_P, RGB_M_B, RGB_M_SW, RGB_M_G, RGB_M_SN,                      XXXXXXX,  XXXXXXX,  XXXXXXX,  XXXXXXX,  KC_COMMA, GAME, 
+    RGB_MOD, RGB_SAI, RGB_SPI, XXXXXXX,  RGB_M_T, CC_ALRM,                       KC_PAST,  KC_P7,    KC_P8,    KC_P9,    KC_PPLS,  KC_PERC,
+    XXXXXXX, RGB_SAD, RGB_SPD, XXXXXXX,  XXXXXXX, XXXXXXX,                       KC_PSLS,  KC_P4,    KC_P5,    KC_P6,    KC_PMNS,  KC_EQL, 
+    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,  KC_NUM,  XXXXXXX, CC_SAVRGB,  _______,  KC_P0,    KC_P1,    KC_P2,    KC_P3,    KC_DOT,   KC_PENT, 
+                               XXXXXXX,  _______, _______, _______,    _______,  _______,  _______,  _______
+),
+
+  [_GAME] = LAYOUT(
+    KC_ESC,  KC_1,  KC_2,  KC_3,    KC_4,    KC_5,                             KC_6,     KC_7,   KC_8,    KC_9,   KC_0,     GAME, 
+    KC_TAB,  KC_Q,  KC_W,  KC_E,    KC_R,    KC_T,                             KC_Z,     KC_U,   KC_I,    KC_O,   KC_P,     KC_PGUP, 
+    KC_LSFT, KC_A,  KC_S,  KC_D,    KC_F,    KC_G,                             KC_H,     KC_J,   KC_K,    KC_L,   KC_F6,    KC_PGDN, 
+    KC_LCTL, KC_Y,  KC_X,  KC_C,    KC_V,    KC_B,    GMESCN,         KC_F12,  KC_N,     KC_M,   KC_LEFT, KC_UP,  KC_RIGHT, KC_DOWN, 
+                           KC_CAPS, KC_LALT, KC_SPC,  KC_SPC,         KC_ENT,  KC_BSPC,  KC_F7,  KC_DEL
+)
+
+};
+
+/* -------------------------------------------------------------------------------- */
+
+/* Different tapping timings */
+uint16_t get_tapping_term(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case GUI_S:
+        case GUI_L:
+            return TAPPING_TERM + 150;
+        case SHFT_SP:
+            return TAPPING_TERM - 50;
+        default:
+            return TAPPING_TERM;
+    }
+}
+
+/* -------------------------------------------------------------------------------- */
+
+/* Rotary encoder */
+#ifdef ENCODER_ENABLE
```

Comment:
Most of this can be replaced with encoder map support.

https://docs.qmk.fm/#/feature_encoders?id=encoder-map